server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -196,30 +197,43 @@ public final class IndicesRequestCache implements RemovalListener<ICacheKey<Indi
-                .setValueType(BytesReference.class)
-                .setKeyType(Key.class)
-                .setRemovalListener(this)
-                .setMaxSizeInBytes(sizeInBytes) // for backward compatibility
-                .setExpireAfterAccess(expire) // for backward compatibility
-                .setDimensionNames(List.of(INDEX_DIMENSION_NAME, SHARD_ID_DIMENSION_NAME))
-                .setCachedResultParser((bytesReference) -> {
-                    try {
-                        return CachedQueryResult.getPolicyValues(bytesReference);
-                    } catch (IOException e) {
-                        // Set took time to -1, which will always be rejected by the policy.
-                        return new CachedQueryResult.PolicyValues(-1);
-                    }
-                })
-                .setKeySerializer(new IRCKeyWriteableSerializer())
-                .setValueSerializer(new BytesReferenceSerializer())
-                .setClusterSettings(clusterService.getClusterSettings())
-                .setStoragePath(nodeEnvironment.nodePaths()[0].path.toString() + "/request_cache")
-                .build(),
-            CacheType.INDICES_REQUEST_CACHE
-        );
+
+        CacheConfig<Key, BytesReference> config = getCacheConfig(settings, nodeEnvironment);
+        this.cache = cacheService.createCache(config, CacheType.INDICES_REQUEST_CACHE);
+    }
+
+    // pkg-private for testing
+    CacheConfig<Key, BytesReference> getCacheConfig(Settings settings, NodeEnvironment nodeEnvironment) {
+        long sizeInBytes = INDICES_CACHE_QUERY_SIZE.get(settings).getBytes();
+        ToLongBiFunction<ICacheKey<Key>, BytesReference> weigher = (k, v) -> k.ramBytesUsed(k.key.ramBytesUsed()) + v.ramBytesUsed();
+        CacheConfig.Builder<Key, BytesReference> configBuilder = new CacheConfig.Builder<Key, BytesReference>().setSettings(settings)
+            .setWeigher(weigher)
+            .setValueType(BytesReference.class)
+            .setKeyType(Key.class)
+            .setRemovalListener(this)
+            .setExpireAfterAccess(expire) // for backward compatibility
+            .setDimensionNames(List.of(INDEX_DIMENSION_NAME, SHARD_ID_DIMENSION_NAME))
+            .setCachedResultParser((bytesReference) -> {
+                try {
+                    return CachedQueryResult.getPolicyValues(bytesReference);
+                } catch (IOException e) {
+                    // Set took time to -1, which will always be rejected by the policy.
+                    return new CachedQueryResult.PolicyValues(-1);
+                }
+            })
+            .setKeySerializer(new IRCKeyWriteableSerializer())
+            .setValueSerializer(new BytesReferenceSerializer())
+            .setClusterSettings(clusterService.getClusterSettings())
+            .setStoragePath(nodeEnvironment.nodePaths()[0].path.toString() + "/request_cache");
+
+        if (!CacheService.pluggableCachingEnabled(CacheType.INDICES_REQUEST_CACHE, settings)) {
```

Comment:
We will be removing pluggable cache feature flag once we go GA(probably in next release). 
Can we mark the old max size setting as deprecated for a start? So that no one uses it. After which we can remove the feature flag and this old setting altogether.